### ISSUE TITLE
Python3 syntax for *args notation of base classes raises an assertion in Nuitka

### ISSUE THREAD

**Issue by pomidoroshev** [kayhayen/Nuitka#75]
Versions:
* Nuitka 0.5.29.4
* Python 3.6.5
* MacOS 10.13.4

Installed Nuitka with `python3 -m pip install nuitka`.

### Problem

`bases.py`:

```python
class A: pass
class B: pass

bases = (A, B)

class C(*bases): pass
```

Verbose output:

```
% python3 -m nuitka --verbose bases.py
Nuitka:WARNING:Problem at '<SourceCodeReference to bases.py:6>' with Starred(value=Name(id='bases', ctx=Load()), ctx=Load()).
Nuitka:WARNING:Problem at '<SourceCodeReference to bases.py:6>' with ClassDef(name='C', bases=[Starred(value=Name(id='bases', ctx=Load()), ctx=Load())], keywords=[], body=[Pass()], decorator_list=[]).
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/nuitka/__main__.py", line 213, in <module>
    main()
  File "/usr/local/lib/python3.6/site-packages/nuitka/__main__.py", line 207, in main
    MainControl.main()
  File "/usr/local/lib/python3.6/site-packages/nuitka/MainControl.py", line 758, in main
    filename = filename
  File "/usr/local/lib/python3.6/site-packages/nuitka/MainControl.py", line 78, in createNodeTree
    is_main  = not Options.shallMakeModule()
  File "/usr/local/lib/python3.6/site-packages/nuitka/tree/Building.py", line 1100, in buildModuleTree
    is_main     = is_main
  File "/usr/local/lib/python3.6/site-packages/nuitka/tree/Building.py", line 1044, in createModuleTree
    is_main     = is_main
  File "/usr/local/lib/python3.6/site-packages/nuitka/tree/Building.py", line 748, in buildParseTree
    source_ref = source_ref
  File "/usr/local/lib/python3.6/site-packages/nuitka/tree/TreeHelpers.py", line 351, in buildStatementsNode
    statements = buildNodeList(provider, nodes, source_ref, allow_none = True)
  File "/usr/local/lib/python3.6/site-packages/nuitka/tree/TreeHelpers.py", line 305, in buildNodeList
    entry = buildNode(provider, node, node_source_ref, allow_none)
  File "/usr/local/lib/python3.6/site-packages/nuitka/tree/TreeHelpers.py", line 262, in buildNode
    source_ref = source_ref
  File "/usr/local/lib/python3.6/site-packages/nuitka/tree/ReformulationClasses.py", line 920, in buildClassNode
    return _buildClassNode3(provider, node, source_ref)
  File "/usr/local/lib/python3.6/site-packages/nuitka/tree/ReformulationClasses.py", line 334, in _buildClassNode3
    source_ref
  File "/usr/local/lib/python3.6/site-packages/nuitka/tree/TreeHelpers.py", line 305, in buildNodeList
    entry = buildNode(provider, node, node_source_ref, allow_none)
  File "/usr/local/lib/python3.6/site-packages/nuitka/tree/TreeHelpers.py", line 276, in buildNode
    assert False, ast.dump(node)
AssertionError: Starred(value=Name(id='bases', ctx=Load()), ctx=Load())
```

**Comment by kayhayen** [comment]
The syntax is of course for all of Python3. 

The "Starred" ast node does not occur with Python 3.4, which seems to be buggy about this:

```
>>> ast.dump(ast.parse("class A(*b): pass"))
"Module(body=[ClassDef(name='A', bases=[], keywords=[], starargs=Name(id='b', ctx=Load()), kwargs=None, body=[Pass()], decorator_list=[])])"
```

With bases simply being empty. This is from Python3.5 or higher:

```
>>> ast.dump(ast.parse("class A(*b): pass"))
"Module(body=[ClassDef(name='A', bases=[Starred(value=Name(id='b', ctx=Load()), ctx=Load())], keywords=[], body=[Pass()], decorator_list=[])])"
```

Nuitka already has handlings for assignments, calls, and sequences for "Starred", and it seems, the bases value definition needs to be treated, most probably, like a sequence creation. If what I see is correct, namely that "bases" is first building the elements, then a sequence from in, instead of creating a "sequence" directly from the list, then it ought to be trivial.

Since I am about to release soon, I am not sure if I am going to make a hotfix out of it, but probably yes.

Yours,
Kay

**Comment by pomidoroshev** [comment]
Hello, Kay, thanks for your response!

I've found this test in `/tests/CPython36/test/test_metaclass.py`:

```python
    >>> class A: pass
    >>> class B: pass
    >>> bases = (A, B)
    >>> class C(*bases): pass
    >>> C.__bases__ == bases
    True
    >>>
```

What triggers this test: Nuitka compiler or CPython interpreter?

**Comment by kayhayen** [comment]
Thanks for looking this up, I was kind of confused how this would silently start to work without a test added, so they did.

Nuitka has a script to generate from doc strings small programs that do the same things, then compares the outputs as usual. I added test_metaclass for 3.6 right now, and it's passing.

When a new test suite is added, I should make an effort to find new doctests, and update this, but I have not taken the time yet. I once did this for say 2.7 and then migrated mostly only.

**Comment by pomidoroshev** [comment]
Oh, I got it.

I thought, that `./tests/CPython36/run_all.py search test_metaclass` runs `/tests/CPython36/test/test_metaclass.py`, but it runs doctests from Python builtin module `test.test_metaclass`, which I didn't know about.

Also, Nuitka successfully compiles and runs this:
```python
import doctest

def foo():
    """
    >>> class A: pass
    >>> class B: pass
    >>> class C(*(A, B)): pass
    >>> C.__bases__ == (A, B)
    True
    """

doctest.testmod(verbose=True)
```

But I don't know anything about `doctest` implementation. Maybe, it uses CPython parser (`eval` or something).

**Comment by kayhayen** [comment]
Yes, indeed, it is using eval, and therefore effectively not testing Nuitka for real. That is why there is a script that parses docstrings and creates code from it, e.g. like this:

http://nuitka.net/gitweb/?p=tests-Nuitka-CPython.git;a=blob_plain;f=doctest_generated/test_metaclass.py;hb=refs/heads/CPython36

Essentially just output an exception if any for every statement and output everything that is an expression. With this simple approach, I am getting this covered too.

The list of doctests is manually maintained though.

Anyway, I did the hotfix, there was something for Windows to be fixed so I did one still.
